junit-jupiter-params/src/main/java/org/junit/jupiter/params/ArgumentCountValidator.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.params;
+
+import static org.junit.platform.commons.support.AnnotationSupport.findAnnotation;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.junit.platform.commons.util.Preconditions;
+
+class ArgumentCountValidator implements InvocationInterceptor {
+	private static final Logger logger = LoggerFactory.getLogger(ArgumentCountValidator.class);
+
+	static final String ARGUMENT_COUNT_VALIDATION_KEY = "junit.jupiter.params.argumentCountValidation";
+
+	private final Arguments arguments;
+
+	ArgumentCountValidator(Arguments arguments) {
+		this.arguments = arguments;
+	}
+
+	@Override
+	public void interceptTestTemplateMethod(InvocationInterceptor.Invocation<Void> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext) throws Throwable {
+		validateArgumentCount(extensionContext, arguments);
+		invocation.proceed();
+	}
+
+	private ExtensionContext.Store getStore(ExtensionContext context) {
+		return context.getRoot().getStore(ExtensionContext.Namespace.create(getClass()));
```

Comment:
Please create a constant for `ExtensionContext.Namespace.create(ArgumentCountValidator.class)`.